File: Cs/Solution/ReverseDegree.cs
namespace Solution
{
	public partial class Solution
	{
		public int ReverseDegree(string s)
		{
			int ret = 0;
			for (int i = 0; i < s.Length; i++)
			{
				ret += (26 + 'a' - s[i]) * (i + 1);
			}
			return ret;
		}
	}
}

File: Rust/src/solution/reverse_degree.rs
use super::Solution;

impl Solution {
    pub fn reverse_degree(s: String) -> i32 {
        s.chars().enumerate().fold(0, |acc, (i, c)| {
            acc + (26 - (c as u8 - b'a')) as i32 * (i as i32 + 1)
        })
    }
}
